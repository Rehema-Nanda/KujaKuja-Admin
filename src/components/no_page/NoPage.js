import React from "react";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowLeft, faArrowRight } from "@fortawesome/free-solid-svg-icons";
import { Row, Col, Container } from "reactstrap";
import PropTypes from "prop-types";

import "./noPage.css";

const facePath = require("../../img/face_unsatisfied.png");

class NoPage extends React.Component {
    goBack = () => {
        const { history } = this.props;

        history.goBack();
    }

    render() {
        return (
            <div>
                <div className="no-page">
                    <div className="title-background-parent">
                        <div className="title-background-shape" />
                    </div>
                    <Container>
                        <Row>
                            <Col sm="12" md={{ size: 6, offset: 3 }}>
                                <div className="no-page-container">
                                    <div className="no-page-icon">
                                        <img src={facePath} alt="alt" />
                                    </div>
                                    <div className="no-page-content">
                                        <h2>404 - Not found</h2>
                                        <p>This error has been recorded and we are looking into it. Please go back and try again or start over from our home page.</p>
                                    </div>
                                    <div className="no-page-buttons">
                                        <Row>
                                            <Col xs="6" className="no-page-option">
                                                <span className="no-page-back" onClick={this.goBack}>
                                                    <FontAwesomeIcon icon={faArrowLeft} />
                                                    {" "}
                                                    Go Back
                                                </span>
                                            </Col>
                                            <Col xs="6" className="no-page-option">
                                                <Link to="/">
                                                    <span className="no-page-back">
                                                        Home
                                                        {" "}
                                                        <FontAwesomeIcon icon={faArrowRight} />
                                                    </span>
                                                </Link>
                                            </Col>
                                        </Row>
                                    </div>
                                </div>
                            </Col>
                        </Row>
                    </Container>
                </div>
            </div>
        );
    }
}

NoPage.propTypes = {
    history: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default NoPage;
