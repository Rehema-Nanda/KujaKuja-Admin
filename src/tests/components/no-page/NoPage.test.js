import React from "react";
import { mount } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import NoPage from "../../../components/no_page/NoPage";

describe("NoPage component", () => {
    const testProps = {
        history: {
            goBack: jest.fn(),
        },
    };
    const wrapper = mount(<BrowserRouter><NoPage {...testProps} /></BrowserRouter>);

    it("render correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("should go back when go back option is clicked", () => {
        const option = wrapper.find(".no-page-back").first();
        expect(option.text()).toEqual(" Go Back");

        option.simulate("click");
        expect(testProps.history.goBack).toHaveBeenCalled();
    });
});
