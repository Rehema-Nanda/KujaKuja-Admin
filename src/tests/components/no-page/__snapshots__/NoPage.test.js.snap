// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoPage component render correctly 1`] = `
"<BrowserRouter>
  <Router history={{...}}>
    <NoPage history={{...}}>
      <div>
        <div className=\\"no-page\\">
          <div className=\\"title-background-parent\\">
            <div className=\\"title-background-shape\\" />
          </div>
          <Container tag=\\"div\\">
            <div className=\\"container\\">
              <Row tag=\\"div\\">
                <div className=\\"row\\">
                  <Col sm=\\"12\\" md={{...}} tag=\\"div\\" widths={{...}}>
                    <div className=\\"col-sm-12 col-md-6 offset-md-3\\">
                      <div className=\\"no-page-container\\">
                        <div className=\\"no-page-icon\\">
                          <img src=\\"face_unsatisfied.png\\" alt=\\"alt\\" />
                        </div>
                        <div className=\\"no-page-content\\">
                          <h2>
                            404 - Not found
                          </h2>
                          <p>
                            This error has been recorded and we are looking into it. Please go back and try again or start over from our home page.
                          </p>
                        </div>
                        <div className=\\"no-page-buttons\\">
                          <Row tag=\\"div\\">
                            <div className=\\"row\\">
                              <Col xs=\\"6\\" className=\\"no-page-option\\" tag=\\"div\\" widths={{...}}>
                                <div className=\\"no-page-option col-6\\">
                                  <span className=\\"no-page-back\\" onClick={[Function]}>
                                    <FontAwesomeIcon icon={{...}}>
                                      <i className=\\"fa\\" />
                                    </FontAwesomeIcon>
                                     
                                    Go Back
                                  </span>
                                </div>
                              </Col>
                              <Col xs=\\"6\\" className=\\"no-page-option\\" tag=\\"div\\" widths={{...}}>
                                <div className=\\"no-page-option col-6\\">
                                  <Link to=\\"/\\" replace={false}>
                                    <a onClick={[Function]} href=\\"/\\">
                                      <span className=\\"no-page-back\\">
                                        Home
                                         
                                        <FontAwesomeIcon icon={{...}}>
                                          <i className=\\"fa\\" />
                                        </FontAwesomeIcon>
                                      </span>
                                    </a>
                                  </Link>
                                </div>
                              </Col>
                            </div>
                          </Row>
                        </div>
                      </div>
                    </div>
                  </Col>
                </div>
              </Row>
            </div>
          </Container>
        </div>
      </div>
    </NoPage>
  </Router>
</BrowserRouter>"
`;
